src/java.base/share/classes/java/lang/constant/DirectMethodHandleDescImpl.java

```diff
@@ -64,7 +64,7 @@ final class DirectMethodHandleDescImpl implements DirectMethodHandleDesc {
      */
     DirectMethodHandleDescImpl(Kind kind, ClassDesc owner, String name, MethodTypeDesc type) {
         if (kind == CONSTRUCTOR)
-            name = "<init>";
+            name = owner.isPrimitiveValueType() ? "<vnew>" : "<init>";
```

Comment:
So here is an issue: `owner.isPrimitiveValueType()` simply says owner is Q-type descriptor. Need this name for Value Objects (and Q-Types), but Value Objects are L-types...

How do we differentiate here ? By using `kind` ?

Reply:
https://cr.openjdk.java.net/~dlsmith/jep8277163/jep8277163-20220830/specs/value-objects-jvms.html suggests REF_invokeStatic